SG14/ring.h

```diff
@@ -10,21 +10,22 @@ namespace sg14
 	template <typename T>
 	struct null_popper
 	{
-		void operator()(T&);
+		void operator()(T&) const noexcept;
 	};
 
 	template <typename T>
 	struct default_popper
 	{
-		T operator()(T& t);
+		T operator()(T& t) const;
 	};
 
 	template <typename T>
 	struct copy_popper
 	{
-		copy_popper(T&& t);
-		T operator()(T& t);
-		T copy;
+		explicit copy_popper(T t);
```

Comment:
Notice that I'm changing this from `T&&` to just `T`, which fixes #102 in an elegant way. I don't think there's any downside to this change.